src/workerd/api/data-url.c++

```diff
@@ -21,12 +22,16 @@ kj::Maybe<DataUrl> DataUrl::from(const jsg::Url& url) {
   // Per the fetch spec, it doesn't matter if the comma is within a quoted
   // string value in the MIME type... which is fun.
 
-  static const auto isAsciiWhitespace = [](auto c) {
-    return c == 0x09 /* tab */ ||
-          c == 0x0a /* lf  */ ||
-          c == 0x0c /* ff  */ ||
-          c == 0x0d /* cr  */ ||
-          c == 0x20 /* sp  */;
+  static constexpr std::array<uint8_t, 256> ascii_whitespace_table = []() consteval {
+    std::array<uint8_t, 256> result{};
+    for (uint8_t c : { 0x09, 0x0a, 0x0c, 0x0d, 0x20 }) {
+      result[c] = true;
+    }
+    return result;
+  }();
+
+  static constexpr auto isAsciiWhitespace = [](uint8_t c) {
+    return ascii_whitespace_table[c];
```

Comment:
LGTM, but out of curiosity couldn't you just do something like:

```suggestion
  static constexpr auto isAsciiWhitespace = [](uint8_t c) {
   if constexpr (c == 0x09 || c== 0x0a blah blah)
```

Reply:
Ah no, that wouldn't be as good as the table for runtime arguments

Reply:
Hash map approach is faster because it uses less instructions: https://godbolt.org/z/zavxo36PM